src/pygama/flow/data_loader.py

```diff
@@ -940,35 +980,48 @@ def fill_col_dict(
                 elif isinstance(tier_table[col], Table):
                     if col not in col_dict.keys():
                         col_dict[col] = {}
-                    col_dict[col] = fill_col_dict(
-                        tier_table[col], col_dict[col], tcm_idx
+                    col_dict[col], attr_dict[col] = fill_col_dict(
+                        tier_table[col], col_dict[col], attr_dict[col], tcm_idx
                     )
                 else:
                     log.warning(
                         f"not sure how to handle column {col} "
                         f"of type {type(tier_table[col])} yet"
                     )
-            return col_dict
+            return col_dict, attr_dict
 
-        def dict_to_table(col_dict: dict):
+        def dict_to_table(col_dict: dict, attr_dict: dict):
             for col in col_dict.keys():
                 if isinstance(col_dict[col], list):
                     if isinstance(col_dict[col][0], (list, np.ndarray, Array)):
-                        col_dict[col] = VectorOfVectors(listoflists=col_dict[col])
+                        col_dict[col] = VectorOfVectors(
+                            listoflists=col_dict[col], attrs=attr_dict[col]
+                        )
                     else:
                         nda = np.array(col_dict[col])
-                        col_dict[col] = Array(nda=nda)
+                        col_dict[col] = Array(nda=nda, attrs=attr_dict[col])
                 elif isinstance(col_dict[col], dict):
-                    col_dict[col] = dict_to_table(col_dict=col_dict[col])
+                    col_dict[col] = dict_to_table(
+                        col_dict=col_dict[col], attr_dict=attr_dict[col]
+                    )
                 else:
                     nda = np.array(col_dict[col])
                     if len(nda.shape) == 2:
-                        col_dict[col] = ArrayOfEqualSizedArrays(nda=nda)
+                        dt = attr_dict[col]["datatype"]
+                        dims = dt[dt.index("<") + 1 : dt.index(">")]
+                        dims = [int(e) for e in dims.split(",")]
```

Comment:
Also nice to use a regex here, something like `re.match("\w+<(\d+),(\d+)>{\w+}", dt)` (untested)

Reply:
Ah sorry, I should have mentioned that the advantage is that by calling `.groups()` on the matched object you can get directly a tuple `('m', 'n')` (your `dims`)